test/junit/scala/tools/nsc/backend/jvm/analysis/TypeFlowAnalyzerTest.scala

```diff
@@ -50,13 +50,14 @@ class TypeFlowAnalyzerTest extends BytecodeTesting {
 
   @Test
   def aaloadTypeNoMerge(): Unit = {
-    // two aaloads: the loaded type is `Object`
     val m = compileAsmMethod("def f(a: Array[String], b: Array[String]) = { if (a eq b) a(0) else b(0) }")
     val a = new NonLubbingTypeFlowAnalyzer(m, "C")
-    val v = a.frameAt(findInstr(m, "ARETURN")).getStack(0)
-    assertEquals(v.getType, BasicValue.REFERENCE_VALUE.getType)
-    assert(!v.isInstanceOf[AaloadValue])
-    assertEquals(a.preciseAaloadTypeDesc(v), "Ljava/lang/Object;")
+    for (instr <- findInstrs(m, "ARETURN")) {
+      val v = a.frameAt(instr).getStack(0)
+      assertEquals(v.getType, BasicValue.REFERENCE_VALUE.getType)
+      assert(v.isInstanceOf[AaloadValue])
+      assertEquals(a.preciseAaloadTypeDesc(v), "Ljava/lang/String;")
```

Comment:
This change from `java.lang.Object` to `java.lang.String` is a progression on this particular example. However, I suspect that this change may void the original intent of this test case. @lrytz Could you take a particular look here?